### PR TITLE
Implement circuit breakers

### PR DESCRIPTION
**Pull request by containscafeine**
- [x] docs
- [x] merge conflicts

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -309,37 +309,41 @@ setup-develop: venv $(KAT_CLIENT) $(TELEPROXY) $(KUBERNAUT) version
...
-	@echo "Done"
+	@echo "Waiting for teleproxy to start..."
+	@sleep 5
+	@if ! pgrep -x "teleproxy" > /dev/null; then \
+		echo "Teleproxy crashed, exiting..."; \
+		exit 1; \
+	fi
+	@echo "Started teleproxy"
+
+teleproxy-restart: teleproxy-stop teleproxy-start
```
Comment (ark3) [inline Makefile]: These will run in arbitrary order

Comment (containscafeine) [reply Makefile]: Hmm. Are you sure? I've been running this target quite a bit and it has been working for me just fine. :confused:

**Comment by containscafeine** [comment]
@kflynn the PR is ready for review. I had one question though. I remember you told something about putting the `circuit_breaker` field under `defaults` field for Ambassador module, but I didn't find any field by that name. Can you clarify once again, thanks ;)

**Comment by kflynn** [comment]
@containscafeine Sorry! We need to add the `defaults` field -- it doesn't exist now, correct.

File: Makefile (hunk 1 above)
Comment (kflynn) [reply Makefile]: Which "these" do you mean, @ark3?

**Comment by containscafeine** [comment]
@kflynn okay, do you want to do that in another PR where we move other fields under default as well?

File: ambassador/ambassador/ir/irambassador.py [hunk 2]
```diff
@@ -36,7 +36,8 @@ class IRAmbassador (IRResource):
         'x_forwarded_proto_redirect',
         'load_balancer',
         'xff_num_trusted_hops',
-        'server_name'
+        'server_name',
+        'circuit_breakers'
```
Comment (kflynn) [inline ambassador/ambassador/ir/irambassador.py]: Once upon a time, these were sorted. They should be sorted again. :)

File: ambassador/ambassador/ir/irhttpmappinggroup.py [hunk 3]
```diff
@@ -43,7 +43,8 @@ class IRHTTPMappingGroup (IRBaseMappingGroup):
         'timeout_ms': True,
         'idle_timeout_ms': True,
         'bypass_auth': True,
-        'load_balancer': True
+        'load_balancer': True,
+        'circuit_breakers': True
```
Comment (kflynn) [inline ambassador/ambassador/ir/irhttpmappinggroup.py]: These were sorted too, once upon a time...

**Comment by kflynn** [comment]
@containscafeine Yeah -- let's talk about that offline, actually.

File: ambassador/ambassador/ir/irambassador.py (hunk 2 above)
Comment (containscafeine) [reply ambassador/ambassador/ir/irambassador.py]: Fixed :+1:

File: ambassador/ambassador/ir/irhttpmappinggroup.py (hunk 3 above)
Comment (containscafeine) [reply ambassador/ambassador/ir/irhttpmappinggroup.py]: Sorted :+1:

File: Makefile (hunk 1 above)
Comment (ark3) [reply Makefile]: Stop and start *may* run in arbitrary order.

Comment (kflynn) [reply Makefile]: Dealt with in `master` and merged in here.

**Pull request merged**